packages/doenetml-worker-rust/lib-doenetml-core/src/components/special/_external.rs

```diff
@@ -1,15 +1,45 @@
 //! An external component is a component not recognized by DoenetML. It is left untouched by the DoenetML processor and returned (mostly)
 //! as-is when converted into `FlatDast`.
 
-use crate::{components::prelude::*, props::UpdaterObject};
+use crate::{
+    components::prelude::*, general_prop::RenderedChildrenPassthroughProp, props::UpdaterObject,
+};
 
 #[derive(Debug, Default, Clone)]
 pub struct _External {
     pub name: String,
 }
 
-#[derive(Debug, Default)]
-pub struct _ExternalProps {}
+impl _External {
+    pub const ATTRIBUTE_NAMES: &'static [&'static str] = &[];
+    const PROPS: &'static [_ExternalProps] = &[_ExternalProps::RenderedChildren];
+    pub const PROP_NAMES: &'static [&'static str] = &["renderedChildren"];
+    const PROP_PROFILES: &'static [Option<PropProfile>] = &[Some(PropProfile::RenderedChildren)];
+    const PROP_FOR_RENDERS: &'static [bool] = &[false];
+    const PROP_IS_PUBLICS: &'static [bool] = &[false];
+    const PROP_VALUE_TYPES: &'static [PropValueType] = &[PropValueType::ContentRefs];
+    const DEFAULT_PROP: Option<LocalPropIdx> = None;
+}
+
+///The props for this component are: `renderedChildren`
+#[derive(Debug, Clone, Copy)]
+pub enum _ExternalProps {
+    /**
+    - Name: `"renderedChildren"`
```

Comment:
This comment is to indicate what the setting would be if the props were created from a macro? Do we want to just go ahead and derive `_External` using the macro? It would seem easier to maintain. I can't remember if there is other special case treatment needed that precludes the use of the macro.

Reply:
I generated from the macro and copy-pasted. We cannot use the macro because the name isn't static.

Reply:
Oh, right, that was the special case.